ext/snmp/snmp.stub.php

```diff
@@ -52,20 +52,20 @@ function snmp3_walk(
     string $hostname, string $security_name, string $security_level,
     string $auth_protocol, string $auth_passphrase,
     string $privacy_protocol, string $privacy_passphrase,
-    array|string $object_id, int $timeout = -1, int $retries = -1): array|bool {}
+    array|string $object_id, int $timeout = -1, int $retries = -1): array|false {}
```

Comment:
I'm simply not able to track down a lot of return types (e.g. this one). The manual writes: "Returns SNMP object value on success or false on error.", so I'm not sure what to believe: the description or the return existing type declaration.

Reply:
It should probably use the same type as snmp2_get and snmpget. I don't think the return type changes based on version.

Reply:
Though seeing https://github.com/php/php-src/blob/787db3af484a4680b2aa25bd1dcc6a9b09f2c1f2/ext/snmp/snmp.c#L314 I think that the return type also needs to include null, which leaves us with `stdClass|array|string|bool|null`, or maybe `stdClass|array|string|false|null` if true is not possible. Maybe it would be better to go with `mixed` for the get functions?

Reply:
Especially as in many cases we actually return integers or floats as strings right now. Unlikely as it is (given how nobody maintains the extension), it may be reasonable for it to return native types in the future...

Reply:
Thanks for checking this! OK, let's go with `mixed` that seems like a future-proof and safe solution. I'm also ok to omit the declaration of tentative return types by only fixing the PHPDoc, if we have no plans with this extension.

Reply:
Type changes look good to me now, please land them on PHP-8.0. I think it's okay to add tentative types as well, for the sake of consistency. At least a rarely used extension also means a rarely extended extension :)

Reply:
Alright, I've already put up a PR for PHP 8.0: https://github.com/php/php-src/pull/7068 I'll merge it a bit later